File: frontend/src/types/itinerary.ts
export interface TransportSegment {
  from_station: string;
  from_station_ja: string;
  to_station: string;
  to_station_ja: string;
  train_name: string;
  train_type: string;
  departure_time: string;
  arrival_time: string;
  duration_minutes: number;
  fare_yen: number;
  covered_by_jr_pass: boolean;
  seat_type?: string;
  platform_note?: string;
  notes?: string;
}

export interface TransportToNext {
  mode: string;
  duration_minutes: number;
  description: string;
  cost_yen?: number;
}

export interface Activity {
  name: string;
  name_ja: string;
  address: string;
  address_ja: string;
  category: string;
  start_time: string;
  end_time: string;
  duration_minutes: number;
  estimated_cost_yen?: number;
  description: string;
  tips?: string;
  seasonal_note?: string;
  nearest_station: string;
  nearest_station_ja: string;
  walking_minutes_from_station: number;
  google_maps_url?: string;
  business_hours?: string;
  phone?: string;
  photo_url?: string;
  highlight?: string;
  transport_to_next?: TransportToNext;
}

export interface DayPlan {
  day_number: number;
  date: string;
  city: string;
  city_ja: string;
  theme: string;
  activities: Activity[];
  transport_segments: TransportSegment[];
  daily_transport_cost_yen: number;
  daily_estimated_total_yen: number;
  weather_note?: string;
  morning_tip?: string;
}

export interface JRPassRecommendation {
  recommended: boolean;
  pass_type: string;
  pass_cost_yen: number;
  individual_tickets_cost_yen: number;
  savings_yen: number;
  covered_segments: string[];
  not_covered_segments: string[];
  recommendation_note: string;
}

export interface Itinerary {
  trip_id: string;
  title: string;
  title_ja?: string;
  start_date: string;
  end_date: string;
  num_days: number;
  cities: string[];
  days: DayPlan[];
  jr_pass_recommendation: JRPassRecommendation;
  total_estimated_cost_yen: number;
  seasonal_highlights: string[];
  packing_tips: string[];
  general_tips: string[];
}

File: backend/app/models/itinerary.py
from datetime import date

from pydantic import BaseModel


class TransportSegment(BaseModel):
    from_station: str
    from_station_ja: str
    to_station: str
    to_station_ja: str
    train_name: str
    train_type: str  # shinkansen | limited_express | local | subway | bus
    departure_time: str
    arrival_time: str
    duration_minutes: int
    fare_yen: int
    covered_by_jr_pass: bool
    seat_type: str = "non-reserved"
    platform_note: str | None = None
    notes: str | None = None


class TransportToNext(BaseModel):
    mode: str  # walk | metro | bus | train | taxi
    duration_minutes: int
    description: str
    cost_yen: int | None = None


class Activity(BaseModel):
    name: str
    name_ja: str
    address: str
    address_ja: str
    category: str  # sightseeing | food | shopping | experience
    start_time: str
    end_time: str
    duration_minutes: int
    estimated_cost_yen: int | None = None
    description: str
    tips: str | None = None
    seasonal_note: str | None = None
    nearest_station: str
    nearest_station_ja: str
    walking_minutes_from_station: int
    google_maps_url: str | None = None
    business_hours: str | None = None
    phone: str | None = None
    photo_url: str | None = None
    highlight: str | None = None
    transport_to_next: TransportToNext | None = None


class DayPlan(BaseModel):
    day_number: int
    date: date
    city: str
    city_ja: str
    theme: str
    activities: list[Activity]
    transport_segments: list[TransportSegment]
    daily_transport_cost_yen: int
    daily_estimated_total_yen: int
    weather_note: str | None = None
    morning_tip: str | None = None


class JRPassRecommendation(BaseModel):
    recommended: bool
    pass_type: str
    pass_cost_yen: int
    individual_tickets_cost_yen: int
    savings_yen: int
    covered_segments: list[str]
    not_covered_segments: list[str]
    recommendation_note: str
    alternative_passes: list[dict] | None = None


class Itinerary(BaseModel):
    trip_id: str
    title: str
    title_ja: str | None = None
    start_date: date
    end_date: date
    num_days: int
    cities: list[str]
    days: list[DayPlan]
    jr_pass_recommendation: JRPassRecommendation
    total_estimated_cost_yen: int
    seasonal_highlights: list[str]
    packing_tips: list[str]
    general_tips: list[str]
